csfieldguide/static/interactives/data-visualisation/js/data-visualisation.js

```diff
@@ -0,0 +1,497 @@
+// const  = CAPS
+// global = CamelCase (start upper)
+// local  = camelCase (start lower)
+
+const CHART = require('chart.js');
+
+const MIN = 0;      // Numbers for data
+const MAX = 9;
+const WAIT = 5000;  // Time (milliseconds) to show the chart
+const BASE_DATA_POINTS = 16;
+const MAX_DATA_POINTS = 256;
+const CHART_TYPES = {
+  GRID: gettext("Plain grid"),
+  MAP:  gettext("Heatmap"),
+  PIE:  gettext("Pie chart"),
+  BAR:  gettext("Bar chart"),
+};
+const TITLES = {    // Bar/Pie chart titles
+  PIE:  gettext("Relative frequency of each number (%)."),
+  BAR:  gettext("Occurences of each number."),
+  RESULTS: gettext("Your accuracy (%) when finding the mode of data with different visualisations.")
+};
+const COLOURS = [   // Pie chart colours
+  "#3e95cd", "#8e5ea2", "#3cba9f", "#e8c3b9", "#c45850",
+  "#28a745","#2a3da0","#34fcfe","#fb8532","#a37533",
+];
+const HEATMAP = [   // Heatmap colours
+  "#0006bf", "#2500c3", "#5200c7", "#8200cb", "#b400cf",
+  "#d400bf","#d80092","#dc0063","#e00032","#e50000",
+];
+
+var CurrentChart;   // Chart to be rendered
```

Comment:
No biggy but vars should be camelCase

Reply:
I did this purposefully for readability, see the [comments I left in](https://github.com/uccser/cs-field-guide/pull/1183#discussion_r342277313)

Reply:
I found the way that the Google JavaScript style guide suggests to define global variables. It can be found [here](https://google.github.io/styleguide/javascriptguide.xml?showone=Naming#Naming) under namespaces.

> ALWAYS prefix identifiers in the global scope with a unique pseudo namespace related to the project or library. If you are working on "Project Sloth", a reasonable pseudo namespace would be sloth.*.

```
var sloth = {};

sloth.sleep = function() {
  ...
};
```